### PR TITLE
Merge branch 'dart-dev' into 'master'.

### LINKED ISSUE #153: Merge dart-dev into master
With the Dart 2.12 beta soon moving to Dart 2.12 stable, we should merge our dart-dev branch (which is on beta) into the master branch.

https://github.com/dart-lang/ffigen/compare/dart-dev?expand=1

We can keep dart-dev around if we'd like to continue working on supporting `dart:ffi` features as they get available on dev. (Maybe we should delete the dart-master branch? If I recall correctly, running tests on master requires building the whole Dart SDK on the Ci just to run the tests.)

### PR DESCRIPTION
**Pull request by mannprerak2**
Closes #153

### CODE REVIEW

File: .github/workflows/test-package.yml [hunk 1]
```diff
@@ -14,7 +14,6 @@ env:
 
 jobs:
   # Check code formatting and static analysis on a single OS (linux)
```
Comment (dcharkes) [inline .github/workflows/test-package.yml:16]: Send sentence with "." ?

**Review by dcharkes — APPROVED** [review]
lgtm

File: .github/workflows/test-package.yml (hunk 1 above)
Comment (mannprerak2) [reply .github/workflows/test-package.yml]: Done. (I don't know why I always keep forgetting this 🙈)

**Comment by dcharkes** [comment]
I prefer to not squash this one, because we already squashed the individual feature PRs on the dart-dev branch.

@mit-mit @athomas any objections?

**Comment by mit-mit** [comment]
I don't have a lot of context here, so hard to comment. The preferred solution is a set of small targeted PRs, each of which can then be squashed.

**Comment by dcharkes** [comment]
We've been extending this package based on new `dart:ffi` features being available on the Dart dev and beta releases on this branch. The master branch has been behind (no support for structs by value, nested structs, breaking API changes etc.)

We've made `package:ffi` pre-releases from this branch. If we squash and merge, we lose the source history for those releases. (I presume the pre-releases themselves are backed up on pub.dev itself.)

File: .github/workflows/test-package.yml [hunk 2]
```diff
@@ -3,24 +3,23 @@ name: Dart CI
 on:
   # Run on PRs and pushes to the default branch.
   push:
-    branches: [ master ]
+    branches: [ dart-dev, master ]
```
Comment (mit-mit) [inline .github/workflows/test-package.yml:6]: Do we really want to run CI on this branch?

**Comment by mit-mit** [comment]
Could we bring the changes over in a set of feature-based PRs?

File: .github/workflows/test-package.yml (hunk 2 above)
Comment (mannprerak2) [reply .github/workflows/test-package.yml:6]: We have decided to keep the `dart-dev` branch for features that are only available on the dev SDK.

We can remove this, but I think it's better to keep it, or we'd have to change this line every time we pull changes between `maser` and `dart-dev` branch.

Comment (dcharkes) [reply .github/workflows/test-package.yml:6]: We want to have a solution for the common situation that:
1) We have a stable version of the package that uses the last Dart stable release and someone makes a PR for that.
2) We have a dev version of the package that uses the last Dart dev release (including new dart:ffi features) and someone makes a PR for that.

So, I think we need two branches.

**Comment by dcharkes** [comment]
We could, but this would be a recurring issue every stable release.

1. We have a stable version of the package that uses the last Dart stable release and someone makes a PR for that.
2. We have a dev version of the package that uses the last Dart dev release (including new dart:ffi features) and someone makes a PR for that.

And then every stable release, we would have to redo all the PRs from the dev branch to the stable branch. Wouldn't it be cleaner to have just a merge from dev to stable every Dart stable release?

**Comment by mannprerak2** [comment]
@dcharkes  Should I squash the last 2 commits into 1 commit?

**Comment by dcharkes** [comment]
Alternative approach as suggested by @mit is to have the master branch track dart dev releases and do all the PRs there. And then when we make a stable release make the master branch track dart stable (or beta).

As for getting this work on to the master branch, we could make individual PRs grouping commits on this branch.

* Support for Nested structs (#133) …
* Support for passing/returning Structs by value. (#134) …
* Cleanups PR
  * Remove setup phase (#135)
  * Removed the usage of --no-sound-null-safety (#137) …
* Globals (#139)
* Use Opaque to represent empty Structs (#142) …
   * (should be part of whatever PR the tests should have been updated in) Updated tests (#143) …
* Skip inline functions, added test, updated changelog and version. (#147)
* Stable release PR
  * Upgrade dependencies (#148)
  * Out of experimental …
  * Bump version. (#151) …
  * Merge branch 'master' of github.com:dart-lang/ffigen into dart-dev

**Comment by dcharkes** [comment]
After discussing with @natebosch, we came up with this: https://github.com/dart-lang/ffigen/wiki/Branches.

**Comment by dcharkes** [comment]
Done in separate PRs.

**Pull request closed without merging**